src/main/java/com/melon/portfoliomanager/dtos/UserDto.java

```diff
@@ -0,0 +1,72 @@
+package com.melon.portfoliomanager.dtos;
+
+
+import jakarta.persistence.Column;
+import jakarta.validation.constraints.Email;
+import jakarta.validation.constraints.NotBlank;
+import jakarta.validation.constraints.NotNull;
+
+public class UserDto {
+
+    @NotNull(message = "Invalid user input. The username is null.")
+    @Column(name = "username", unique = true)
+    private String username;
+
+    @Email(message = "Invalid user input. The email should be valid.")
+    @NotNull(message = "Invalid user input. The email is null.")
```

Comment:
Are you sure that ```@Email``` annotation does not cover the case when the email is null ? (and this way the ```@NotNull``` is redundant may be)

Reply:
Both are needed, null elements are considered valid with the email annotation. :)